scripts/contacts/migrateContactsV1toV2.js

```diff
@@ -0,0 +1,167 @@
+const has = require('lodash/has')
+
+const mkAPI = require('../api')
+const {
+  migrationDiff: diff,
+  getWithInstanceLogger
+} = require('../../libs/utils')
+
+const CONTACTS_APP_NAME = 'Contacts'
+const KONNECTOR_APP_NAME = 'konnector-google'
+const VCARD_APP_NAME = 'vcard'
+const ACCOUNT_APP_VERSION = 1
+const DOCTYPE_CONTACTS = 'io.cozy.contacts'
+const DOCTYPE_ACCOUNTS = 'io.cozy.accounts'
+const DOCTYPE_CONTACTS_ACCOUNT = 'io.cozy.contacts.accounts'
+const DOCTYPE_CONTACTS_VERSION = 2
+
+let client, api
+
+const getCreatedByApp = contact => {
+  const metadataKeys = Object.keys(contact.metadata).filter(
+    k => k !== 'version'
+  )
+
+  if (metadataKeys.includes('google')) {
+    return KONNECTOR_APP_NAME
+  } else if (metadataKeys.includes('cozy')) {
+    return CONTACTS_APP_NAME
+  } else if (metadataKeys.length === 0) {
+    return VCARD_APP_NAME
+  }
+
+  return null
+}
+
+const migrateContactV1toV2 = (contact, accountId, contactAccountId) => {
+  const now = new Date().toISOString()
+  const createdByApp = getCreatedByApp(contact)
+  let accounts = []
+  let sourceAccount, sync
+  if (contactAccountId && createdByApp === KONNECTOR_APP_NAME) {
+    sync = {
+      [contactAccountId]: {
+        contactsAccountsId: contactAccountId,
+        id: contact.vendorId,
+        konnector: KONNECTOR_APP_NAME,
+        lastSync: now,
+        remoteRev: null
+      }
+    }
+    sourceAccount = accountId
+    accounts = [
+      {
+        _id: contactAccountId,
+        _type: DOCTYPE_CONTACTS_ACCOUNT
+      }
+    ]
+  }
+
+  return {
+    ...contact,
+    me: false,
+    cozyMetadata: {
+      sync,
+      createdAt: null,
+      createdByApp,
+      doctypeVersion: DOCTYPE_CONTACTS_VERSION,
+      sourceAccount,
+      updatedAt: now,
+      updatedByApps: [createdByApp]
+    },
+    relationships: {
+      ...contact.relationships,
+      accounts: {
+        data: accounts
+      }
+    }
+  }
+}
+
+const migrateContactsV1toV2 = (contacts, accountId, contactAccountId) =>
```

Comment:
I suggest inlining the call to `.map` in the main function. When we call `migrateContactsV1toV2 ` we can't really now if the array returned will have the same length or if there will be some filtering or whatnot.